core/mm/core_mmu.c

```diff
@@ -110,6 +120,53 @@ void core_mmu_get_secure_memory(paddr_t *base, paddr_size_t *size)
 	*size = secure_only[0].size;
 }
 
+#ifdef CFG_CORE_RELOCATABLE
+void core_mmu_set_secure_memory(paddr_t base, size_t size)
+{
+	static_assert(ARRAY_SIZE(secure_only) == 1);
+	assert(!secure_only[0].size);
+	assert(base && size);
+
+	DMSG("Physical secure memory base %#"PRIxPA" size %#zx", base, size);
+	secure_only[0].paddr = base;
+	secure_only[0].size = size;
+}
+#endif
+
+void core_mmu_get_ta_range(paddr_t *base, size_t *size)
+{
+	paddr_t b = 0;
+	size_t s = 0;
+
+	static_assert(!(TEE_RAM_VA_SIZE % SMALL_PAGE_SIZE));
+#ifdef TA_RAM_START
+	b = TA_RAM_START;
+	s = TA_RAM_SIZE;
+#else
+	static_assert(ARRAY_SIZE(secure_only) <= 2);
+	if (ARRAY_SIZE(secure_only) == 1) {
+		vaddr_t load_offs = 0;
+
+		assert(core_mmu_tee_load_pa >= secure_only[0].paddr);
+		load_offs = core_mmu_tee_load_pa - secure_only[0].paddr;
+
+		assert(secure_only[0].size >
+		       load_offs + TEE_RAM_VA_SIZE + TEE_SDP_TEST_MEM_SIZE);
+		b = secure_only[0].paddr + load_offs + TEE_RAM_VA_SIZE;
+		s = secure_only[0].size - load_offs - TEE_RAM_VA_SIZE -
```

Comment:
It seems like the lower part of the secure memory (below `core_mmu_tee_load_pa`) is not being taken account into the TA RAM. Am I right? If so, do you plan to submit patches to also take account the lower memory?

Reply:
That part is currently expected to be quite small, 16kB or so in a Hafnium setup. I haven't made up my mind about what to do with that yet.